### PR TITLE
[server] Fix org git auth

### PR DESCRIPTION
**Pull request by easyCZ**
## Related Issue(s)

Fixes #

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - mp-fix-git-provider</li>
	<li><b>🔗 URL</b> - <a href="https://mp-fix-git-provider.preview.gitpod-dev.com/workspaces" target="_blank">mp-fix-git-provider.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [x] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

File: components/server/src/user/user-service.ts [hunk 1]
```diff
@@ -262,25 +262,33 @@ export class UserService {
...
     async updateUserIdentity(user: User, candidate: Identity) {
+        log.info("Updating user identity", {
+            user,
+            candidate,
+        });
         // ensure single identity per auth provider instance
         user.identities = user.identities.filter((i) => i.authProviderId !== candidate.authProviderId);
         user.identities.push(candidate);
-
-        await this.userDb.storeUser(user);
```
Comment (AlexTugarev) [inline components/server/src/user/user-service.ts:283]: why removing this? doesn't this break `updateUserOnLogin`?

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 2]
```diff
@@ -675,7 +675,7 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         );
 
         if (this.user) {
-            await this.userToTeamMigrationService.migrateUser(this.user, true, "on demand");
+            await this.userToTeamMigrationService.migrateUser(this.user.id, true, "on demand");
```
Comment (AlexTugarev) [inline components/server/src/workspace/gitpod-server-impl.ts:678]: 🎉

File: components/server/src/user/user-service.ts (hunk 1 above)
Comment (easyCZ) [reply components/server/src/user/user-service.ts:283]: Not sure how the change came about. We can re-add

Comment (easyCZ) [reply components/server/src/user/user-service.ts]: re-added

File: components/server/src/user/user-service.ts [hunk 3]
```diff
@@ -262,20 +262,30 @@ export class UserService {
...
+        }
+        return updated;
     }
 
     async onAfterUserLoad(user: User): Promise<User> {
         return user;
     }
 
     async updateUserIdentity(user: User, candidate: Identity) {
+        log.info("Updating user identity", {
```
Comment (AlexTugarev) [inline components/server/src/user/user-service.ts:285]: Q: is it intended to be on info level? that might be very noisy.

**Review by AlexTugarev — APPROVED** [review]

**Pull request merged**